lists/parsers/hosts.go

```diff
@@ -10,14 +10,21 @@ import (
 	"regexp"
 	"strings"
 
-	"github.com/asaskevich/govalidator"
 	"github.com/hashicorp/go-multierror"
 	"golang.org/x/net/idna"
 )
 
-const maxDomainNameLength = 255 // https://www.rfc-editor.org/rfc/rfc1034#section-3.1
+const (
+	maxDomainNameLength = 255 // https://www.rfc-editor.org/rfc/rfc1034#section-3.1
 
-var domainNameRegex = regexp.MustCompile(govalidator.DNSName)
+	dnsLabelPattern = `[a-zA-Z0-9_-]{1,63}`
+)
+
+// Validate a domain name, but with extra flexibility:
+// - no restriction on the start or end of labels
+//
+// https://www.rfc-editor.org/rfc/rfc1034#section-3.5
+var domainNameRegex = regexp.MustCompile(`^` + dnsLabelPattern + `(\.` + dnsLabelPattern + `)*[\._]?$`)
```

Comment:
Not sure what the `_` is for in the regex. I just copied it from [govalidator](https://github.com/asaskevich/govalidator/blob/v11.0.1/patterns.go#L33).  
It doesn't seem relevant to me, maybe I should remove it?

Reply:
Wikipedia says:
"While a hostname may not contain other characters, such as the underscore character (_), other DNS names may contain the underscore.[7][8] This restriction was lifted by RFC 2181. Systems such as DomainKeys and service records use the underscore as a means to assure that their special character is not confused with hostnames."
https://en.m.wikipedia.org/wiki/Hostname

Therefore I think it should stay.

Reply:
Thanks, makes sense!